centreon/src/Core/HostTemplate/Application/UseCase/AddHostTemplate/AddHostTemplate.php

```diff
@@ -407,4 +236,119 @@ private function findParentTemplates(array $templateIds): array
 
         return $parentTemplates;
     }
+
+    /**
+     * @param AddHostTemplateRequest $dto
+     * @param int $hostTemplateId
+     *
+     * @throws \Throwable
+     */
+    private function addMacros(AddHostTemplateRequest $dto, int $hostTemplateId): void
+    {
+        $this->info(
+            'AddHostTemplate: Add macros',
+            ['host_template_id' => $hostTemplateId, 'macros' => $dto->macros]
+        );
+
+        /**
+         * @var array<string,HostMacro> $inheritedMacros
+         * @var array<string,CommandMacro> $commandMacros
+         */
+        [$inheritedMacros, $commandMacros]
+            = $this->findAllInheritedMacros($hostTemplateId, $dto->checkCommandId);
+
+        $macros = [];
+        foreach ($dto->macros as $data) {
+            $macro = HostMacroFactory::create($data, $hostTemplateId, $inheritedMacros);
+            $macros[$macro->getName()] = $macro;
+        }
+
+        $macrosDiff = new HostMacroDifference();
+        $macrosDiff->compute([], $inheritedMacros, $commandMacros, $macros);
+
+        MacroManager::setOrder($macrosDiff, $macros, []);
+
+        foreach ($macrosDiff->addedMacros as $macro) {
+            if ($macro->getDescription() === '') {
+                $macro->setDescription(
+                    isset($commandMacros[$macro->getName()])
+                    ? $commandMacros[$macro->getName()]->getDescription()
+                    : ''
+                );
+            }
+            $this->writeHostMacroRepository->add($macro);
+        }
+
+    }
+
+    /**
+     * Find macros of a host template:
+     * macros linked through template inheritance, macros linked through command inheritance.
+     *
+     * @param int $hostTemplateId
+     * @param int $checkCommandId
+     *
+     * @throws \Throwable
+     *
+     * @return array{
+     *      array<string,HostMacro>,
+     *      array<string,CommandMacro>
+     * }
+     */
+    private function findAllInheritedMacros(int $hostTemplateId, ?int $checkCommandId): array
+    {
+        $templateParents = $this->readHostTemplateRepository->findParents($hostTemplateId);
+        $inheritanceLine = HostInheritance::findInheritanceLine($hostTemplateId, $templateParents);
+        $existingHostMacros = $this->readHostMacroRepository->findByHostIds($inheritanceLine);
+
+        [$directMacros, $inheritedMacros] = MacroManager::resolveInheritanceForHostMacro(
+            $existingHostMacros,
+            $inheritanceLine,
+            $hostTemplateId
+        );
+
+        /** @var array<string,CommandMacro> */
```

Comment:
Don't forget variable, official syntax is `/** @var MyType $variable */`
```suggestion
        /** @var array<string,CommandMacro> $commandMacros */
```